DataFormats/HLTReco/interface/TriggerRefsCollections.h

```diff
@@ -754,7 +836,81 @@ namespace trigger
       }
       return;
     }
-
+    void getObjects(int id, VRl1tmuon& refs) const {
+      getObjects(id,refs,0,l1tmuonIds_.size());
+    } 
+    void getObjects(int id, VRl1tmuon& refs, size_type begin, size_type end) const {
+      assert (begin<=end);
+      assert (end<=l1tmuonIds_.size());
+      size_type n(0);
+      for (size_type i=begin; i!=end; ++i) {if (id==l1tmuonIds_[i]) {++n;}}
+      refs.resize(n);
+      size_type j(0);
+      for (size_type i=begin; i!=end; ++i) {
+	if (id==l1tmuonIds_[i]) {refs[j]=l1tmuonRefs_[i]; ++j;}
+      }
+      return;
+    }
+    void getObjects(int id, VRl1tegamma& refs) const {
+      getObjects(id,refs,0,l1tegammaIds_.size());
+    } 
+    void getObjects(int id, VRl1tegamma& refs, size_type begin, size_type end) const {
+      assert (begin<=end);
+      assert (end<=l1tegammaIds_.size());
+      size_type n(0);
+      for (size_type i=begin; i!=end; ++i) {if (id==l1tegammaIds_[i]) {++n;}}
+      refs.resize(n);
+      size_type j(0);
+      for (size_type i=begin; i!=end; ++i) {
+	if (id==l1tegammaIds_[i]) {refs[j]=l1tegammaRefs_[i]; ++j;}
+      }
+      return;
+    }
+    void getObjects(int id, VRl1tjet& refs) const {
+      getObjects(id,refs,0,l1tjetIds_.size());
+    } 
+    void getObjects(int id, VRl1tjet& refs, size_type begin, size_type end) const {
+      assert (begin<=end);
+      assert (end<=l1tjetIds_.size());
+      size_type n(0);
+      for (size_type i=begin; i!=end; ++i) {if (id==l1tjetIds_[i]) {++n;}}
+      refs.resize(n);
+      size_type j(0);
+      for (size_type i=begin; i!=end; ++i) {
+	if (id==l1tjetIds_[i]) {refs[j]=l1tjetRefs_[i]; ++j;}
+      }
+      return;
+    }
+    void getObjects(int id, VRl1ttau& refs) const {
+      getObjects(id,refs,0,l1ttauIds_.size());
+    } 
+    void getObjects(int id, VRl1ttau& refs, size_type begin, size_type end) const {
+      assert (begin<=end);
+      assert (end<=l1ttauIds_.size());
+      size_type n(0);
+      for (size_type i=begin; i!=end; ++i) {if (id==l1ttauIds_[i]) {++n;}}
+      refs.resize(n);
+      size_type j(0);
+      for (size_type i=begin; i!=end; ++i) {
+	if (id==l1ttauIds_[i]) {refs[j]=l1ttauRefs_[i]; ++j;}
+      }
+      return;
+    }
+    void getObjects(int id, VRl1tetsum& refs) const {
+      getObjects(id,refs,0,l1tetsumIds_.size());
+    } 
+    void getObjects(int id, VRl1tetsum& refs, size_type begin, size_type end) const {
+      assert (begin<=end);
+      assert (end<=l1tetsumIds_.size());
+      size_type n(0);
+      for (size_type i=begin; i!=end; ++i) {if (id==l1tetsumIds_[i]) {++n;}}
+      refs.resize(n);
+      size_type j(0);
+      for (size_type i=begin; i!=end; ++i) {
+	if (id==l1tetsumIds_[i]) {refs[j]=l1tetsumRefs_[i]; ++j;}
+      }
+      return;
+    }
     void getObjects(Vids& ids, VRpfjet& refs) const {
       getObjects(ids,refs,0,pfjetIds_.size());
     }
```

Comment:
You also need to add the low-level accessors which are here:
  https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/DataFormats/HLTReco/interface/TriggerRefsCollections.h#L854